packages/amplify-category-auth/resources/auth-custom-resource/openIdLambda.js

```diff
@@ -1,36 +1,77 @@
-          .promise()
-          .catch((err) => {
-            console.log(err);
-
-            if (err.name === 'NotFoundException') {
-              return response.send(event, context, response.SUCCESS);
-            }
-
-            response.send(event, context, response.FAILED, { err });
-          });
-      }
-    })
-    .catch((err) => {
-      console.log(err);
-      response.send(event, context, response.FAILED, { err });
-    });
+exports.handler = (event, context) => {
+  if (event.RequestType == 'Delete') {
+    response.send(event, context, response.SUCCESS, {});
+  }
+  if (event.RequestType == 'Update' || event.RequestType == 'Create') {
+    const params = {
+      ClientIDList: event.ResourceProperties.clientIdList.split(','),
+      ThumbprintList: ['0000000000000000000000000000000000000000'],
+      Url: event.ResourceProperties.url,
+    };
+    let exists = false;
+    let existingValue;
+    iam
+      .listOpenIDConnectProviders({})
+      .promise()
+      .then((data) => {
+        if (data.OpenIDConnectProviderList && data.OpenIDConnectProviderList.length > 0) {
+          const vals = data.OpenIDConnectProviderList.map((x) => x.Arn);
+          existingValue = vals.find((i) => i.split('/')[1] === params.Url.split('//')[1]);
+          if (!existingValue) {
+            exists = true;
+          }
+        }
+        if (!existingValue) {
+          iam
+            .createOpenIDConnectProvider(params)
+            .promise()
+            .then((data) => {
+              response.send(event, context, response.SUCCESS, {
+                providerArn: data.OpenIDConnectProviderArn,
+                providerIds: params.ClientIDList,
+              });
+            })
+            .catch((err) => {
+              response.send(event, context, response.FAILED, { err });
+            });
+        } else {
+          const findParams = {
+            OpenIDConnectProviderArn: existingValue,
+          };
+          iam
+            .getOpenIDConnectProvider(findParams)
+            .promise()
+            .then((data) => {
+              const audiences = data.ClientIDList;
+              const updateCalls = [];
+              params.ClientIDList.forEach((a) => {
+                if (!audiences.includes(a)) {
+                  const updateParams = {
+                    ClientID: a,
+                    OpenIDConnectProviderArn: existingValue,
+                  };
+                  const prom = iam.addClientIDToOpenIDConnectProvider(updateParams).promise();
+                  updateCalls.push(prom);
+                }
+              });
+              Promise.all(updateCalls)
+                .then(function (values) {
+                  response.send(event, context, response.SUCCESS, { providerArn: existingValue, providerIds: params.ClientIDList });
+                })
+                .catch((err3) => {
+                  response.send(event, context, response.FAILED, { err3 });
```

Comment:
For this line, 69, and 74, this is sending `{ err3: err3 }`. Is that what is wanted? Or do we need to send `{ err: err3 }`? Why name the errors err1, err2, err3, etc, if the variable name is not shadowed?